### ISSUE TITLE
Rule Change: Mark only the function name instead of all of the function lines

### ISSUE THREAD

**Issue by moshest** [eslint/eslint#16023]
### What rule do you want to change?

max-lines-per-function

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* 01 */ /*eslint max-lines-per-function: ["error", 2]*/
/* 02 */ function foo() {
/* 03 */   // a comment followed by a blank line
/* 04 */ 
/* 05 */   var x = 0;
/* 06 */ }
/* 07 */ 
/* 08 */ const bar = () => {
/* 09 */   // a comment followed by a blank line
/* 10 */ 
/* 11 */   var x = 0;
/* 12 */ };
```

### What does the rule currently do for this code?

It mark all the function lines as error (lines `02-06` and `08-12`)

### What will the rule do after it's changed?

It will mark only the function header as an error. 

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

The reason for that is to allow the developer to easily read the function code and refactor it. The current state, that all the lines mark as error/warning make it very hard to edit the function and fix the problem.

**Comment by mdjermanovic** [comment]
Hi @moshest, thanks for the issue! 

> It will mark only the function open and close brackets `{` `}` as an error.

Each error/warning marks a continuous range, so we can't mark only the brackets. We could consider marking the first line that counts or something like that (e.g., function header) instead of the whole function, but marking only `{` and `}` isn't doable.

**Comment by moshest** [comment]
Function header sound great, I've update the description.

**Comment by moshest** [comment]
Btw, the rule "max-lines" (for files) mark only the lines that exceed the limit. maybe we can have the same solution here.

**Comment by moshest** [comment]
I'm using this rule for a few months now. I think it will be really helpful to warn only the function header and not all the body.

The code is not readable when all the function body is in warning and it's hard to refactor!

**Comment by nzakas** [comment]
There’s not enough support on the team to move forward, so closing.
